app/src/main/java/com/github/sdpteam15/polyevents/model/Event.kt

```diff
@@ -56,55 +55,41 @@ data class Event(
     }
 
     /**
-     * Set the amount of a certain item in the inventory of the event
-     * @param item: the item to add
-     * @param amount the amount of the item to add.
-     * @return the old previous amount of that item, or null if the item was not found
-     */
-    fun setItemAmount(item: Item, amount: Int): Int? {
-        if (amount < 0) {
-            throw IllegalArgumentException("Amount of items must >= 0")
-        } else {
-            return inventory.put(item, amount)
-        }
-    }
+     * Add an item to the event's inventory
+     * @param item the item to add
+     * @return true if the item was successfully added
+      */
+    fun addItem(item: Item): Boolean =
+        inventory.add(item)
 
     /**
-     * Add an amount of an item to the inventory of the event
-     * @param item: the item to add
-     * @param amount the amount of the item to add. By default is 1
-     * @return the old previous amount of that item, or null if the item was not found
+     * Remove item from the event's inventory
+     * @param item the item to remove
+     * @return true if the item was successfully removed
      */
-    fun addItem(item: Item, amount: Int = 1): Int? {
-        val currentAmount = inventory.getOrDefault(item, 0)
-        return inventory.put(item, currentAmount + amount)
-    }
+    fun removeItem(item: Item): Boolean =
+        inventory.remove(item)
 
     /**
-     * Take an amount of some item from the inventory of the event.
-     *
-     * @param item the item to take
-     * @param amount amount of the item to take. By default is 1
-     * @return the old previous amount of that item, or null if the item was not found
+     * Check if event's inventory has certain item
+     * @param item the item to check
+     * @return true if the item is there
      */
-    fun takeItem(item: Item, amount: Int = 1): Int? {
-        val currentAmount = inventory.getOrDefault(item, 0)
-        if (amount > currentAmount) {
-            throw InsufficientAmountException("There are only $currentAmount of $item available")
-        } else {
-            return inventory.put(item, currentAmount - amount)
-        }
-    }

```

Comment:
Why did you remove this function ?

Reply:
This was the first version when Item didn't have an Id, just an itemType. It was easier to keep track of how many of each item type an event had using a map. But now items are "unique",  so we add them to a list or a set. This can still be changed.

Reply:
Ok, I got it !